tracee/tracee.go

```diff
@@ -0,0 +1,500 @@
+	}
+	return string(res), nil
+}
+
+func readInt8FromBuff(buff io.Reader) (int8, error) {
+	var res int8
+	err := binary.Read(buff, binary.LittleEndian, &res)
+	return res, err
+}
+
+func readUInt8FromBuff(buff io.Reader) (uint8, error) {
+	var res uint8
+	err := binary.Read(buff, binary.LittleEndian, &res)
+	return res, err
+}
+
+func readInt16FromBuff(buff io.Reader) (int16, error) {
+	var res int16
+	err := binary.Read(buff, binary.LittleEndian, &res)
+	return res, err
+}
+
+func readUInt16FromBuff(buff io.Reader) (uint16, error) {
+	var res uint16
+	err := binary.Read(buff, binary.LittleEndian, &res)
+	return res, err
+}
+
+func readInt32FromBuff(buff io.Reader) (int32, error) {
+	var res int32
+	err := binary.Read(buff, binary.LittleEndian, &res)
+	return res, err
+}
+
+func readUInt32FromBuff(buff io.Reader) (uint32, error) {
+	var res uint32
+	err := binary.Read(buff, binary.LittleEndian, &res)
+	return res, err
+}
+
+func readInt64FromBuff(buff io.Reader) (int64, error) {
+	var res int64
+	err := binary.Read(buff, binary.LittleEndian, &res)
+	return res, err
+}
+
+func readUInt64FromBuff(buff io.Reader) (uint64, error) {
+	var res uint64
+	err := binary.Read(buff, binary.LittleEndian, &res)
+	return res, err
+}
+
+func readArgFromBuff(dataBuff io.Reader) (interface{}, error) {
+	var err error
+	var res interface{}
+	at, err := readArgTypeFromBuff(dataBuff)
+	if err != nil {
+		return res, fmt.Errorf("error reading arg type: %v", err)
+	}
+	switch at {
+	case INT_T:
+		res, err = readInt32FromBuff(dataBuff)
+		if err != nil {
+			return nil, err
+		}
+	case UINT_T, DEV_T_T:
+		res, err = readUInt32FromBuff(dataBuff)
+		if err != nil {
+			return nil, err
+		}
+	case LONG_T:
+		res, err = readInt64FromBuff(dataBuff)
+		if err != nil {
+			return nil, err
+		}
+	case ULONG_T, OFF_T_T, SIZE_T_T:
+		res, err = readUInt64FromBuff(dataBuff)
+		if err != nil {
+			return nil, err
+		}
+	case STR_T:
+		res, err = readStringFromBuff(dataBuff)
+		if err != nil {
+			return nil, err
+		}
+	case STR_ARR_T:
+		var ss []string
+		// assuming there's at least one element in the array
+		et, err := readArgTypeFromBuff(dataBuff)
+		if err != nil {
+			return nil, fmt.Errorf("error reading string array element type: %v", err)
+		}
+		for et != STR_ARR_T {
+			s, err := readStringFromBuff(dataBuff)
+			if err != nil {
+				return nil, fmt.Errorf("error reading string element: %v", err)
+			}
+			ss = append(ss, s)
+
+			et, err = readArgTypeFromBuff(dataBuff)
+			if err != nil {
+				return nil, fmt.Errorf("error reading string array element type: %v", err)
+			}
+		}
+		res = ss
+	case CAP_T:
+		cap, err := readInt32FromBuff(dataBuff)
+		if err != nil {
+			return nil, fmt.Errorf("error reading capability arg: %v", err)
+		}
+		res = PrintCapability(cap)
+	case SYSCALL_T:
+		sc, err := readInt32FromBuff(dataBuff)
+		if err != nil {
+			return nil, fmt.Errorf("error reading syscall arg: %v", err)
+		}
+		if e, ok := EventsIDToName[sc]; ok {
```

Comment:
This will not work with your recent changes!
We need to have a full map of syscall ids to syscall names

Reply:
Why do you think this will not work?

Reply:
got it, fixed in 33e946e72f0ba58500b440622c1a7653afe6ade4